sky-uk/cqlmigrate
# Update packages dependencies to latest versions, Gradle version to la…

…test, and tests to use new package versions

- Gradle 5.3
- `java-library` plugin instead of `java` plugin
- Gradle plugin updates
- build script polish
- `functional` task improved to work when Java 11 in on the path (tests do not pass on Java 11)
- use `api` and `implementation` scopes to generate an improved POM
- test dependency updates
- Cassandra driver update to 3.7.1 (going to need a much bigger update for 4.0 https://github.com/datastax/java-driver/blob/8402f53eb4dc96413137ddfe741193ce380f2dc4/upgrade_guide/README.md)
- up-to-date Gradle checks for `functional` execution
- Mockito 2 update
- Test fixes due to various library upgrades

This upgrade should help make it easier to develop with this tool going forward, especially with the recent release of the 4.0 driver.

------

### Notes

* I don't know if publishing is still going to work the right way
* Cassandra driver 4.0 throws a wrench in a lot of the current consumed tools, so it will be a bit more pain to go to that
* This will at least help users with the latest java-driver version
